cpp/src/Slice/Grammar.y

```diff
@@ -565,26 +572,14 @@ tag
     if(constant)
     {
         BuiltinPtr b = BuiltinPtr::dynamicCast(constant->type());
-        if(b)
+        if(b && b->isWholeNumber())
```

Comment:
I think the function should be named `isIntegralNumber`

Reply:
`isInteger` was my first though, but I think it could possibly lead to confusion with the `int` type. I'm fine with `isIntegralNumber`, but now that I stare at the names more, maybe `isIntegralType` might be more sensible... What do you think?

Reply:
`isIntegralType` is fine